c7n/filters/core.py

```diff
@@ -239,6 +239,9 @@ def get_resource_value(self, k, i, regex=None):
             # GCP schema: 'labels': {'key': 'value'}
             elif 'labels' in i:
                 r = i.get('labels', {}).get(tk, None)
+            elif 'settings' in i:
```

Comment:
Rather than modifying the core filter here, we should implement an augment method on the resource itself, see example here: https://github.com/cloud-custodian/cloud-custodian/pull/7534/files#diff-8a6994421549ec47f2ce8f24a8ab517b097e3292b005b485e2e85ae0c581f557R43-R48